src/sync.rs

```diff
@@ -713,6 +729,47 @@ fn test_non_lockfree_unchecked() {
     LockFreeFrozenVec::<()>::new();
 }
 
+impl<T: Copy + Clone> Clone for LockFreeFrozenVec<T> {
+    fn clone(&self) -> Self {
+        let len = self.len.load(Ordering::Acquire);
+        // handle the empty case
+        if len == 0 {
+            return Self::default();
+        }
+
+        // copy the data
+        let data = [Self::NULL; NUM_BUFFERS];
+        // for each buffer, copy the data
+        for i in 0..NUM_BUFFERS {
+            // get the buffer size and index
+            let buffer_size = buffer_size(i);
+            let buffer_idx = buffer_index(buffer_size - 1);
+            // get the buffer pointer
+            let buffer_ptr = self.data[buffer_idx].load(Ordering::Acquire);
+            if buffer_ptr.is_null() {
+                // no data in this buffer
+                break;
+            }
+            // allocate a new buffer
+            let layout = Self::layout(buffer_size);
+            let new_buffer_ptr = unsafe { std::alloc::alloc(layout).cast::<T>() };
+            assert!(!new_buffer_ptr.is_null());
+            // copy the data
+            unsafe {
+                std::ptr::copy_nonoverlapping(buffer_ptr, new_buffer_ptr, buffer_size);
+            }
+            // store the new buffer pointer
+            data[i].store(new_buffer_ptr, Ordering::Release);
```

Comment:
You can use `get_mut` instead of `store`, you have unique access.

Reply:
Fixed in [`aa7ee7a` (#57)](https://github.com/Manishearth/elsa/pull/57/commits/aa7ee7a317d154cc17f0ff05e3549f011376f6c1)